packages/auth-sveltekit/src/server.ts

```diff
@@ -0,0 +1,489 @@
+    }
+
+    return { tokenData: null };
+  }
+
+  async emailPasswordResendVerificationEmail({
+    verificationToken,
+  }: {
+    verificationToken: string;
+  }): Promise<void> {
+    await (await this.core).resendVerificationEmail(verificationToken);
+  }
+
+  async emailPasswordSignIn({
+    email,
+    password,
+  }: {
+    email: string;
+    password: string;
+  }): Promise<{ tokenData?: TokenData; error?: Error }> {
+    const tokenData = await (
+      await this.core
+    ).signinWithEmailPassword(email, password);
+
+    this.cookies.set(this.config.authCookieName, tokenData.auth_token, {
+      httpOnly: true,
+      sameSite: "strict",
+      path: "/",
+    });
+
+    return { tokenData };
+  }
+
+  async emailPasswordSendPasswordResetEmail({
+    email,
+  }: {
+    email: string;
+  }): Promise<void> {
+    if (!this.config.passwordResetPath) {
+      throw new Error(`'passwordResetPath' option not configured`);
+    }
+
+    const { verifier } = await (
+      await this.core
+    ).sendPasswordResetEmail(
+      email,
+      new URL(this.config.passwordResetPath, this.config.baseUrl).toString()
+    );
+
+    this.cookies.set(this.config.pkceVerifierCookieName, verifier, {
+      httpOnly: true,
+      sameSite: "strict",
+      path: "/",
+    });
+  }
+
+  async emailPasswordResetPassword({
+    resetToken,
+    password,
+  }: {
+    resetToken: string;
+    password: string;
+  }): Promise<{ tokenData: TokenData }> {
+    const verifier = this.cookies.get(this.config.pkceVerifierCookieName);
+
+    if (!verifier) {
+      throw new Error("no pkce verifier cookie found");
+    }
+
+    const tokenData = await (
+      await this.core
+    ).resetPasswordWithResetToken(resetToken, verifier, password);
+
+    this.cookies.set(this.config.authCookieName, tokenData.auth_token, {
+      httpOnly: true,
+      sameSite: "lax",
+      path: "/",
+    });
+
+    this.cookies.delete(this.config.pkceVerifierCookieName, {
+      path: "/",
+    });
+    return { tokenData };
+  }
+
+  async signout(): Promise<void> {
+    this.cookies.delete(this.config.authCookieName, { path: "/" });
+  }
+}
+
+export class AuthSession {
+  public readonly client: Client;
+
+  /** @internal */
+  constructor(client: Client, private readonly authToken: string | undefined) {
+    this.client = this.authToken
+      ? client.withGlobals({ "ext::auth::client_token": this.authToken })
+      : client;
+  }
+
+  async isSignedIn() {
+    if (!this.authToken) return false;
+    try {
+      return await this.client.querySingle<boolean>(
+        `select exists global ext::auth::ClientTokenIdentity`
+      );
+    } catch {
+      return false;
+    }
+  }
+}
+
+async function createAuthRouteHandlers(
+  {
+    onOAuthCallback,
+    onBuiltinUICallback,
+    onEmailVerify,
+    onSignout,
+  }: Partial<AuthRouteHandlers>,
+  event: RequestEvent,
+  core: Promise<Auth>,
+  config: AuthConfig
+) {
+  const url = new URL(event.request.url);
```

Comment:
I guess we can check if this is also already a `URL` object?

```suggestion
  const { url } = event.request;
```

Reply:
```
async function handleAuthRoutes(
  ...
  { url, cookies }: RequestEvent,
  ...
) {
  const searchParams = url.searchParams;
  const path = url.pathname.split("/").slice(2).join("/");
  ...
```